interface_tester/interface_test.py

```diff
@@ -379,22 +380,21 @@ def _run(self, event: Union[str, Event]):
         # some required config, a "happy" status, network information, OTHER relations.
         # Typically, should NOT touch the relation that this interface test is about
         #  -> so we overwrite and warn on conflict: state_template is the baseline,
-        state = (self.ctx.state_template or State()).copy()
+        state = copy.deepcopy(self.ctx.state_template or State())
```

Comment:
not sure if it's better but I usually do a `dataclasses.replace(dc_in)` instead of deepcopy, also it's pointless to copy if it's a fresh instance (not your fault, I know...)
```suggestion
        state = dataclasses.replace(self.ctx.state_template) if self.ctx.state_template else State()
```